ledger/participant-integration-api/src/test/suite/scala/platform/store/backend/UpdateToDbDtoSpec.scala

```diff
@@ -0,0 +1,1191 @@
+          submitters = Some(submitterInfo.actAs.toSet),
+          contract_id = exerciseNode.targetCoid.coid,
+          template_id = None,
+          tree_event_witnesses = Set("divulgee"),
+          create_argument = None,
+          create_argument_compression = compressionAlgorithmId,
+          event_sequential_id = 0,
+        ),
+        DbDto.CommandCompletion(
+          completion_offset = someOffset.toHexString,
+          record_time = update.recordTime.toInstant,
+          application_id = submitterInfo.applicationId,
+          submitters = submitterInfo.actAs.toSet,
+          command_id = submitterInfo.commandId,
+          transaction_id = Some(update.transactionId),
+          status_code = None,
+          status_message = None,
+        ),
+      )
+    }
+    "handle TransactionAccepted (no submission info)" in {
+      // Transaction that is missing a SubmitterInfo
+      // This happens if a transaction was submitted through a different participant
+      val transactionMeta = someTransactionMeta
+      val builder = new TransactionBuilder()
+      val createNode = builder.create(
+        id = builder.newCid,
+        template = "pkgid:M:T",
+        argument = Value.ValueUnit,
+        signatories = List("signatory"),
+        observers = List("observer"),
+        key = None,
+      )
+      val createNodeId = builder.add(createNode)
+      val transaction = builder.buildCommitted()
+      val update = Update.TransactionAccepted(
+        optSubmitterInfo = None,
+        transactionMeta = transactionMeta,
+        transaction = transaction,
+        transactionId = TransactionId.assertFromString("TransactionId"),
+        recordTime = someRecordTime,
+        divulgedContracts = List.empty,
+        blindingInfo = None,
+      )
+      val dtos = UpdateToDbDto(someParticipantId, valueSerialization, compressionStrategy)(
+        someOffset
+      )(update).toList
+
+      dtos should contain theSameElementsInOrderAs List(
+        DbDto.EventCreate(
+          event_offset = Some(someOffset.toHexString),
+          transaction_id = Some(update.transactionId),
+          ledger_effective_time = Some(transactionMeta.ledgerEffectiveTime.toInstant),
+          command_id = None,
+          workflow_id = transactionMeta.workflowId,
+          application_id = None,
+          submitters = None,
+          node_index = Some(createNodeId.index),
+          event_id = Some(EventId(update.transactionId, createNodeId).toLedgerString),
+          contract_id = createNode.coid.coid,
+          template_id = Some(createNode.coinst.template.toString),
+          flat_event_witnesses = Set("signatory", "observer"),
+          tree_event_witnesses = Set("signatory", "observer"),
+          create_argument = Some(emptyArray),
+          create_signatories = Some(Set("signatory")),
+          create_observers = Some(Set("observer")),
+          create_agreement_text = Some(createNode.coinst.agreementText).filter(_.nonEmpty),
```

Comment:
I think these kind of replication of the actual code under test is a bit on the non-practical side. would you mind changing this to the actual Option[String] instead?
(applies for similar constructs)